ykpack/src/lib.rs

```diff
@@ -90,40 +78,30 @@ mod tests {
         vec![sir1, sir2]
     }
 
-    // Check serialising and deserialising works for zero packs.
-    #[test]
-    fn test_empty() {
-        let mut curs = get_curs();
-
-        let enc = Encoder::from(&mut curs);
-        enc.done().unwrap();
-
-        rewind_curs(&mut curs);
-        let mut dec = Decoder::from(&mut curs);
-        assert!(dec.next().unwrap().is_none());
-    }
-
     // Check a typical serialising and deserialising session.
     #[test]
     fn test_basic() {
         let inputs = get_sample_packs();
-        let mut curs = get_curs();
-
-        let mut enc = Encoder::from(&mut curs);
+        let num_packs = inputs.len();
+        let mut buf = Vec::new();
+        let mut enc = Encoder::from(&mut buf);
         for md in &inputs {
             enc.serialise(md.clone()).unwrap();
         }
-        enc.done().unwrap();
 
-        rewind_curs(&mut curs);
+        let mut curs = Cursor::new(&mut buf);
         let dec = Decoder::from(&mut curs);
 
         // Obtain two fallible iterators, so we can zip them.
         let expect_iter = fallible_iterator::convert(inputs.into_iter().map(|e| Ok(e)));
 
         let mut itr = dec.zip(expect_iter);
-        while let Some((got, expect)) = itr.next().unwrap() {
+        for _ in 0..num_packs {
+            let (got, expect) = itr.next().unwrap().unwrap();
             assert_eq!(expect, got);
         }
+
+        // We've consumed everything, so attempting to decode another pack should fail.
+        assert!(itr.next().is_err());
```

Comment:
This feels like a `debug_assert` to me. [Similarly the `assert_eq` just above on line 101/126.]

Reply:
This is a test, so it should be `assert!`.

Reply:
Every `assert` is a test ;)

Reply:
Mea culpa! I see what you mean.